axlearn/open_api/metrics/tool_use_execution_utils.py

```diff
@@ -0,0 +1,307 @@
+    "haven't",
+    "having",
+    "he",
+    "her",
+    "here",
+    "hers",
+    "herself",
+    "him",
+    "himself",
+    "his",
+    "how",
+    "i",
+    "if",
+    "in",
+    "into",
+    "is",
+    "isn",
+    "isn't",
+    "it",
+    "it's",
+    "its",
+    "itself",
+    "just",
+    "ll",
+    "m",
+    "ma",
+    "me",
+    "mightn",
+    "mightn't",
+    "more",
+    "most",
+    "mustn",
+    "mustn't",
+    "my",
+    "myself",
+    "needn",
+    "needn't",
+    "no",
+    "nor",
+    "not",
+    "now",
+    "o",
+    "of",
+    "off",
+    "on",
+    "once",
+    "only",
+    "or",
+    "other",
+    "our",
+    "ours",
+    "ourselves",
+    "out",
+    "over",
+    "own",
+    "re",
+    "s",
+    "same",
+    "shan",
+    "shan't",
+    "she",
+    "she's",
+    "should",
+    "should've",
+    "shouldn",
+    "shouldn't",
+    "so",
+    "some",
+    "such",
+    "t",
+    "than",
+    "that",
+    "that'll",
+    "the",
+    "their",
+    "theirs",
+    "them",
+    "themselves",
+    "then",
+    "there",
+    "these",
+    "they",
+    "this",
+    "those",
+    "through",
+    "to",
+    "too",
+    "under",
+    "until",
+    "up",
+    "ve",
+    "very",
+    "was",
+    "wasn",
+    "wasn't",
+    "we",
+    "were",
+    "weren",
+    "weren't",
+    "what",
+    "when",
+    "where",
+    "which",
+    "while",
+    "who",
+    "whom",
+    "why",
+    "will",
+    "with",
+    "won",
+    "won't",
+    "wouldn",
+    "wouldn't",
+    "y",
+    "you",
+    "you'd",
+    "you'll",
+    "you're",
+    "you've",
+    "your",
+    "yours",
+    "yourself",
+    "yourselves",
+}
+
+
+def _string_lenient_transform(str_value: str) -> str:
+    """Performs a lenient string transformation."""
+    normalized_str_value = str_value.lower()
+    normalized_str_value = normalized_str_value.translate(str.maketrans("", "", string.punctuation))
+    words = normalized_str_value.split()
+    while len(words) > 1 and words[0] in _STOP_WORDS:
+        words = words[1:]
+    while len(words) > 1 and words[-1] in _STOP_WORDS:
+        words = words[:-1]
+    normalized_str_value = " ".join(words)
+    return normalized_str_value
+
+
+def _word_set(input_str: str) -> set[str]:
+    """Returns a set of words splitting by whitespace, newline and tab,
+    and skipping empty strings from the set."""
+    return {s for s in re.split(r"\s+", input_str) if s}
+
+
+def _match_strings_bag_of_words(pred_str: str, target_str: str, threshold: float = 1.0) -> bool:
+    """Match strings using a bag of words approach.
+
+    Args:
+        pred_str: The predicted argument string.
+        target_str: The target argument string.
+        threshold: applied to (# unique common words) / (# unique pred_str words)
+
+    Returns:
+        True if recall is higher or equal to threshold; otherwise return False.
+    """
+    if pred_str == target_str:
+        return True
+
+    assert threshold > 0, "bag of words string matching threshold must be above 0"
```

Comment:
```suggestion
    assert threshold > 0, "Bag of words string matching threshold must be above 0."
```
It's preferable to raise a ValueError for caller-provided values. Asserts are usually used for catching logical bugs.

Reply:
Done